iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/IpRestrictedAdminAccessCheck.java

```diff
@@ -39,11 +41,29 @@ public void initialize(InitContext init) {
     init.register(ResourceDeclaration.class, (ctx, resource) -> {
       if (RESOURCE_TYPE.equals(resource.type().value())) {
         PropertyUtils.get(resource, "sourceAddressPrefix").ifPresent(propertyTree -> {
-          if (TextUtils.matchesValue(propertyTree.value(), SOURCE_ADDRESS_PREFIX_SENSITIVE::contains).isTrue()) {
+          if (isSensitiveString(propertyTree.value())) {
+            ctx.reportIssue(propertyTree, MESSAGE);
+          }
+        });
+        PropertyUtils.get(resource, "sourceAddressPrefixes").ifPresent(propertyTree -> {
+          if (isSensitiveArray(propertyTree.value())) {
             ctx.reportIssue(propertyTree, MESSAGE);
           }
         });
       }
     });
   }
+
+  private static boolean isSensitiveString(Tree value) {
```

Comment:
```suggestion
  private static boolean isSensitiveSourceAddress(Tree value) {
```
Make it more explicit what is checked here.